File: codechef/Red_Blue_Flowers.cpp
#include <bits/stdc++.h>
using namespace std;

#define FOR(i,a,b) for(int i=a;i<b;++i)
#define ll long long

const int maxN = 1e3+5;
int red[maxN];
int blue[maxN];

int main(){
    int t;
    cin>>t;
    FOR(_,0,t){
        int n;
        cin>>n;
        ll ans = 0;
        FOR(i,1,n+1) cin>>red[i+1];
        FOR(i,1,n+1) cin>>blue[i+1];

        ll ts=0;
        FOR(i,1,n+1){
            ts += red[i];
        }
        int dp[n+1][ts+1];
        memset(dp,-1,sizeof(dp));
        dp[0][0] = 0;

        FOR(j,0,ts+1)
            dp[0][j] = 0;
        FOR(i,1,n+1)
            dp[i][0] = dp[i-1][0]+blue[i];

        FOR(i,1,n+1){
            FOR(j,1,ts+1){
                if(dp[i][j-red[i]] != -1){
                    
                }
            }
        }
        cout<<ans<<'\n';
    }
}

File: codechef/Red_Blue_Flowers.py
# cook your dish here
from functools import lru_cache
t = int(input())
for _ in range(t):
    n = int(input())
    red = list(map(int,input().split()))
    blue = list(map(int,input().split()))
    ans=0
    
    @lru_cache(1000)
    def dp(i,r,b):
        global ans
        if i == n:
            ans = max(ans,min(r,b))
            return;
        dp(i+1,r+red[i],b)
        dp(i+1,r,b+blue[i])
        
    dp(0,0,0)
    print(ans)
        